src/Kernel.package/CompiledMethod.class/instance/isDeprecated.st

```diff
@@ -1,9 +1,13 @@
 testing
 isDeprecated
-
-	^ self sendsAnySelectorOf: #(
-		#deprecated: 
-		#deprecated:on:in:
-		#deprecated:on:in:transformWith:
-		#deprecated:transformWith: 
-	)
\ No newline at end of file
+	"Object selectorsInProtocol: #deprecation"
+	(self
+		sendsAnySelectorOf:
+			#(#deprecated: #deprecated:on:in: #deprecated:on:in:transformWith: #deprecated:transformWith:))
+		ifTrue: [ ^ true ].
+	$-
+		split: self protocol asString
+		do: [ :each | 
+			each withBlanksCondensed = 'deprecated'
```

Comment:
Actually, we should ask if the method belongs to a deprecated package. That will solve extension methods.